src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -0,0 +1,144 @@
+(ns puppetlabs.services.legacy-routes.legacy-routes-core
+  (:require [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [compojure.core :as compojure]
+            [ring.util.codec :as ring-codec]
+            [compojure.route :as route]
+            [clojure.tools.logging :as log]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; v3 => v4 header translation support functions
+
+; TODO: Improve the naive parsing here with a real library like
+; https://github.com/ToBeReplaced/http-accept-headers
+(defn- map-accept-raw-to-binary
+  "Convert Accept: raw media types to Accept: binary.  NOTE: This method does
+  not conform to RFC-2616 Accept: header specification, but then again
+  neither does Accept: raw or Accept: binary
+
+  The split on `,` is naive, the method should take into account `;` parameters
+  and the such to be RFC compliant, but puppet agent does not use this format
+  so implementing the full spec is unnecessary."
+  [request]
+  (if-let [accept (get-in request [:headers "accept"])]
+    (let [vals (str/split accept #"\s*,\s*")]               ; BAD! See RFC-2616
+      (assoc-in request [:headers "accept"]
+        (str/join ", " (distinct (replace {"raw" "binary"} vals)))))
+    request))
+
+(defn- map-accept-s-pson-to-binary
+  "Convert Accept: s, pson media types to Accept: binary given a ring request
+  map.  This method should only be used to munge file_bucket_file API requests
+  from legacy agents speaking to a contemporary master."
+  [request]
+  (if-let [accept (get-in request [:headers "accept"])]
+    (let [v4-accept-val "binary"
+          vals (str/split accept #"\s*,\s*")                ; BAD! see RFC-2616
+          munged-accept (->> vals
+                          (replace {"s" v4-accept-val "pson" v4-accept-val})
```

Comment:
@camlow325 

> So is pson one that is still valid for some endpoints and not others in Puppet 4, or should we be filtering it out for all endpoints?

It's still there for some requests in Puppet 4, e.g. NODE GET my-agent which looks like:

```
~~~~~ NODE GET my-agent ~~~~~
2015-04-17 12:43:14,908 INFO  [c.p.t.p.CharLoggingHandler] [id: 0x4135ecd2, /127.0.0.1:53807 => /127.0.0.1:8666]
INPUT
GET /puppet/v3/node/my-agent?environment=production&transaction_uuid=539ac6f4-a5ad-4970-ad7c-7c4ac675c08a&fail_on_404=true HTTP/1.1^M
Accept: pson, msgpack, yaml, binary^M
X-Puppet-Version: 4.0.0^M
Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3^M
User-Agent: Ruby^M
Host: localhost:8666^M
^M

2015-04-17 12:43:14,934 INFO  [c.p.t.p.CharLoggingHandler] [id: 0x4135ecd2, /127.0.0.1:53807 => /127.0.0.1:8666]
OUTPUT
HTTP/1.1 200 OK ^M
X-Puppet-Version: 4.0.0^M
Content-Type: text/pson^M
Server: WEBrick/1.3.1 (Ruby/1.9.3/2014-11-13) OpenSSL/0.9.8za^M
Date: Fri, 17 Apr 2015 19:43:14 GMT^M
ontent-Length: 46^M
Connection: Keep-Alive^M
^M
```